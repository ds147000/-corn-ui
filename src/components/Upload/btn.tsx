import { View } from '@tarojs/components'
import React from 'react'
import Icon from '../Icon'
import InputMask from './inputMask'
import { InputMaskProps } from './inputMask/typeing'

export const UploadDefaultBtn: React.FC<InputMaskProps> = (props) => {
  return (
    <View className="corn-upload-btn corn-f corn-ac corn-jc" data-testid="upload-btn">
      <InputMask {...props} />
      <Icon name="img" />
    </View>
  )
}
