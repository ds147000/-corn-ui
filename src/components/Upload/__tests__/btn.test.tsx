import React from 'react'
import { render } from '@testing-library/react'
import { UploadDefaultBtn } from '../btn'

jest.mock('../inputMask')
jest.mock('../../Toast')

test('UploadDefaultBtn', () => {
  const screen = render(<UploadDefaultBtn type="all" multiple onChange={jest.fn()} />)
  expect(screen.container).toMatchSnapshot()
})
