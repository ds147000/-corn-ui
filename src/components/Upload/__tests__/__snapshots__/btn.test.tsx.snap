// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadDefaultBtn 1`] = `
<div>
  <div
    class="corn-upload-btn corn-f corn-ac corn-jc"
    data-testid="upload-btn"
  >
    <input
      accept="video/*,image/*"
      class="corn-upload-mask"
      data-testid="uplaod-btn"
      multiple=""
      type="file"
    />
    <i
      class="corn-icon corn-i-img"
    />
  </div>
</div>
`;
